urla39b63eb76_tef_pipeline_tgz-pJ8-org_twentygototen_pipeline_ExampleJ8/src/org/twentygototen/pipeline/sources/SourceWithPipe.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * 
+ */
+package org.twentygototen.pipeline.sources;
+
+import org.twentygototen.pipeline.Callbacks;
+import org.twentygototen.pipeline.Pipeline;
+import org.twentygototen.pipeline.Callbacks.Callback;
+import org.twentygototen.pipeline.Pipeline.Block;
+import org.twentygototen.pipeline.Pipeline.Sink;
+import org.twentygototen.pipeline.Pipeline.Pipe;
+import org.twentygototen.pipeline.Pipeline.Source;
+
+class SourceWithPipe<I,O> implements Source<O> {
+	private final Pipe<? super I, O> pipe;
+	private final Source<? extends I> source;
+
+	public SourceWithPipe(Source<? extends I> source, Pipe<? super I, O> pipe) {
+		this.source = source;
+		this.pipe = pipe;
+	}
+
+
+	public O emit() {
+		I produce = null;
+		try {
+			produce = source.emit();
+		} catch (RuntimeException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)